src/Microsoft.AspNet.Server.Kestrel.Https/HttpsConnectionFilterOptions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.AspNet.Server.Kestrel.Https
+{
+    public class HttpsConnectionFilterOptions
+    {
+        public HttpsConnectionFilterOptions()
+        {
+            ClientCertificateMode = ClientCertificateMode.NoCertificate;
+            SslProtocols = SslProtocols.Tls12 | SslProtocols.Tls11 | SslProtocols.Tls;
+        }
+
+        public X509Certificate2 ServerCertificate { get; set; }
```

Comment:
@Tratcher Depending on how much flexibility you want to allow, each option could be based on the ServerAddress.
```
UseKestrelHttps(this IApplicationBuilder app, Func<ServerAddress, HttpsConnectionFilterOptions>)
```